test/CadlRanchProjects/models/property-optional/Generated/ModelsPropertyOptionalClientBuilderExtensions.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using Azure.Core.Extensions;
+using Models.Property.Optional;
+
+namespace Microsoft.Extensions.Azure
+{
+    /// <summary> Extension methods to add clients to client builder. </summary>
+    public static partial class ModelsPropertyOptionalClientBuilderExtensions
+    {
+        /// <param name="builder"></param>
+        /// <param name="endpoint"> TestServer endpoint. </param>
+        public static IAzureClientBuilder<OptionalClient, OptionalClientOptions> AddOptionalClient<TBuilder>(this TBuilder builder, Uri endpoint)
+        where TBuilder : IAzureClientFactoryBuilder
+        {
+            return builder.RegisterClientFactory<OptionalClient, OptionalClientOptions>((options) => new OptionalClient(endpoint, options));
+        }
+
+        /// <param name="builder"></param>
+        /// <param name="configuration"></param>
+        public static IAzureClientBuilder<OptionalClient, OptionalClientOptions> AddOptionalClient<TBuilder, TConfiguration>(this TBuilder builder, TConfiguration configuration)
+        where TBuilder : IAzureClientFactoryBuilderWithConfiguration<TConfiguration>
+        {
+            return builder.RegisterClientFactory<OptionalClient, OptionalClientOptions>(configuration);
+        }
+        /// <param name="builder"></param>
+        /// <param name="configuration"></param>
+        public static IAzureClientBuilder<Models.Property.Optional.String, OptionalClientOptions> AddString<TBuilder, TConfiguration>(this TBuilder builder, TConfiguration configuration)
```

Comment:
This looks weird, and I even wonder whether if it works. But I think we could create an issue about the subclient scenarios and figure it out later.

Reply:
Or maybe we just not generate it for now.

Reply:
OperationGroups and subclients should be excluded here.